function-service-impl/src/main/java/com/future/function/service/impl/feature/scoring/AssignmentServiceImpl.java

```diff
@@ -52,9 +59,17 @@ public AssignmentServiceImpl(
       .map(batch -> assignmentRepository.findAllByBatchAndDeletedFalse(batch,
                                                                        pageable
       ))
+      .map(this::sortByClosestDeadline)
```

Comment:
I think this is invalid as the sorting is done after getting paged data.

For example:
12 assignments data, each for batch 3 and `deleted` flag is false. The last data (12th) actually has the closest deadline, but since the query calls only all by `batch` and `deleted` false, only the first 10 is obtained, then those 10 data are sorted, though I believe what should be the desired output is that the 12th data should be among those 10 data (whereas it isn't).

A solution would be to use aggregation pipeline in mongodb, an example can be seen at `ActivityBlogRepositoryImpl`

Reply:
can i index the deadline field so when i fetch it, the 12th data will be there?

Reply:
If so, please create the index as well in this PR. cc @oliverrsebastian , thank you